### ISSUE TITLE
evaluate does not allow you to return a json object

### ISSUE THREAD

**Issue by AndrewRayCode** [sheebz/phantom-proxy#16]
According to the phantomjs docs:

Note: The arguments **and the return value to the evaluate function must be a simple primitive object.** The rule of thumb: **if it can be serialized via JSON, then it is fine.** Closures, functions, DOM nodes, etc. will not work!

Right now, if you return an object in an evaluate function, you get back string `"[object Object]"` which is undesired. JSON results should be deserialized if possible. I'm deep in the bowels of `main.js` and hope to have a PR for this soon.

**Comment by AndrewRayCode** [comment]
ok, well I have no idea where to look, but this is a fine test for it:

```
it('should json deserialize from evaluate', function (done) {
    phantomProxy.create({"debug":true}, function (proxy) {
        should.exist(proxy.page);
        proxy.page.evaluate(function (x) {
            return {res: x};
        }, function (result) {
            assert.equal(result.res, 4);
            done();
        }, 2);
    });
});
```

**Comment by sheebz** [comment]
Yeah, based on api docs, you appear to be correct.  I will look into this.  In the mean time,  you should be able to work around it by calling JSON.stringify(object) in the client side function, then JSON.parse in the callback.

**Comment by ghost** [comment]
@sheebz bump
